compat/src/Children.js

```diff
@@ -17,5 +17,14 @@ export const Children = {
 		if (normalized.length !== 1) throw 'Children.only';
 		return normalized[0];
 	},
-	toArray: toChildArray
+	toArray(children) {
+		return toChildArray(children).map((child, i) => {
+			// Only normalized vnodes will be keyed #2888
+			if (child != null && typeof child === 'object') {
+				child.key = child.key != null ? child.key : i;
```

Comment:
I think we may have to be a little fancier here... check out the note in the react docs: https://reactjs.org/docs/react-api.html#reactchildrentoarray 

Would be curious what the use case is

Reply:
Strong +1 for "what is the use-case for this". I can't think of a situation in which this behavior is actually desirable, even if it's what React currently does.